packages/gatsby-remark-images/src/index.js

```diff
@@ -113,7 +116,7 @@ module.exports = (
     const presentationWidth = fluidResult.presentationWidth
 
     // Generate default alt tag
-    const srcSplit = node.url.split(`/`)
+    const srcSplit = queryString.parseUrl(node.url).url.split(`/`)
```

Comment:
Not specific to your PR (this behavior already existed!), but this seems like a good opportunity for a helper, perhaps something like?

```js
const getImageInfo = uri => {
  const { url, query } = queryString.parseUrl(uri)
  return {
    ext: path.extname(url),
    url,
    query
  }
}
```

[Here's an example](https://runkit.com/dschau/5c6483a2bbdc5c00133343cf)

Reply:
I've added a helper in 185f8ca. The filename without the extension is pulled out for a default alt tag, which could be added to the helper. But that's the only place that's done, so I thought best to leave that part unless it's needed elsewhere. I'm happy to move it into the helper too if wanted though :-)

I'll look into adding a test next.